CONTRIBUTING.md

```diff
@@ -0,0 +1,165 @@
+
+### External contributors
+
+If you are not a member of the `dbt-labs` GitHub organization, you can contribute to `dbt-bigquery` by forking the `dbt-bigquery` repository. For a detailed overview on forking, check out the [GitHub docs on forking](https://help.github.com/en/articles/fork-a-repo). In short, you will need to:
+
+1. fork the `dbt-bigquery` repository
+2. clone your fork locally
+3. check out a new branch for your proposed changes
+4. push changes to your fork
+5. open a pull request against `dbt-labs/dbt-bigquery` from your forked repository
+
+### Core contributors
+
+If you are a member of the `dbt-labs` GitHub organization, you will have push access to the `dbt` repo. Rather than forking `dbt-bigquery` to make your changes, just clone the repository, check out a new branch, and push directly to that branch.
+
+## Setting up an environment
+There are some tools that will be helpful to you in developing locally. While this is the list reveleant for `dbt` development, many of these tools are used commonly across open-source python projects.
+
+### Tools
+
+A short list of tools used in `dbt` testing that will be helpful to your understanding:
+
+- [`tox`](https://tox.readthedocs.io/en/latest/) to manage virtualenvs across python versions. We currently target the latest patch releases for Python 3.6, Python 3.7, Python 3.8, and Python 3.9
+- [`pytest`](https://docs.pytest.org/en/latest/) to discover/run tests
+- [`make`](https://users.cs.duke.edu/~ola/courses/programming/Makefiles/Makefiles.html) - but don't worry too much, nobody _really_ understands how make works and our Makefile is super simple
+- [`flake8`](https://flake8.pycqa.org/en/latest/) for code linting
+- [`mypy`](https://mypy.readthedocs.io/en/stable/) for static type checking
+- [Github Actions](https://github.com/features/actions)
+
+A deep understanding of these tools in not required to effectively contribute to `dbt`, but we recommend checking out the attached documentation if you're interested in learning more about them.
+
+#### virtual environments
+
+We strongly recommend using virtual environments when developing code in `dbt`. We recommend creating this virtualenv
+in the root of the `dbt` repository. To create a new virtualenv, run:
+```sh
+python3 -m venv env
+source env/bin/activate
+```
+
+This will create and activate a new Python virtual environment.
+
+#### docker and docker-compose
+
+Docker and docker-compose are both used in testing. Specific instructions for you OS can be found [here](https://docs.docker.com/get-docker/).
+
+## Running `dbt-bigquery` in development
+
+### Installation
+
+First make sure that you set up your `virtualenv` as described in [Setting up an environment](#setting-up-an-environment).  Also ensure you have the latest version of pip installed with `pip install --upgrade pip`. Next, install `dbt-biguery` latest dependencies:
+
+```sh
+pip install -r dev-requirements.txt
```

Comment:
```suggestion
pip install -e . -r dev-requirements.txt
```
Note that **this is a place where our adapter plugins will differ from dbt-core**. In the `dbt-core` repo, this needs to be:
```
pip install -r editable-requirements.txt -r dev-requirements.txt
```
And in dbt-spark (special case), this needs to include [extras](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies) like so:
```
pip install -e .[all] -r dev-requirements.txt
```